src/pr-info.ts

```diff
@@ -417,21 +425,48 @@ configSuspicious["tsconfig.json"] = makeJsonCheckerFromCore(
 // to it, ignoring some keys (JSON Patch paths).  The ignored properties are in most cases checked
 // elsewhere (dtslint), and in some cases they are irrelevant.
 function makeJsonCheckerFromCore(requiredForm: any, ignoredKeys: string[]) {
-    const diffFromReq = (text: string) => {
-        let json: any;
-        try { json = JSON.parse(text); } catch (e) { return "couldn't parse json"; }
-        jsonDiff.applyPatch(json, ignoredKeys.map(path => ({ op: "remove", path })));
-        try { return jsonDiff.compare(requiredForm, json); } catch (e) { return "couldn't diff json" };
-    };
     return (contents: string, oldText?: string) => {
-        const newDiff = diffFromReq(contents);
-        if (typeof newDiff === "string") return newDiff;
+        const diffFromReq = (json: any) => {
+            jsonDiff.applyPatch(json, ignoredKeys.map(path => ({ op: "remove", path })));
+            return jsonDiff.compare(requiredForm, json);
+        };
+        let newJson;
+        try { newJson = JSON.parse(contents); } catch (e) { return { suspect: "couldn't parse json" }; }
+        const suggestion = jsonDiff.deepClone(newJson);
+        const newDiff = diffFromReq(newJson);
         if (newDiff.length === 0) return undefined;
-        if (!oldText) return "not the required form";
-        const oldDiff = diffFromReq(oldText);
-        if (typeof oldDiff === "string") return oldDiff;
-        if (jsonDiff.compare(oldDiff, newDiff).every(({ op }) => op === "remove")) return undefined;
-        return "not the required form and not moving towards it";
+        const towardsIt = jsonDiff.deepClone(requiredForm);
+        if (oldText) {
+            let oldJson;
+            try { oldJson = JSON.parse(oldText); } catch (e) { return { suspect: "couldn't parse json" }; }
+            const oldDiff = diffFromReq(oldJson);
+            const notRemove = jsonDiff.compare(oldDiff, newDiff).filter(({ op }) => op !== "remove");
+            if (notRemove.length === 0) return undefined;
+            jsonDiff.applyPatch(newDiff, notRemove.map(({ path }) => ({ op: "remove", path })));
+            jsonDiff.applyPatch(towardsIt, newDiff);
+        }
+        jsonDiff.applyPatch(suggestion, jsonDiff.compare(newJson, towardsIt));
+        // Suggest the different lines to the author
+        const suggestionLines = (
+            Object.keys(suggestion).length <= 1
+                ? prettier.format(JSON.stringify(suggestion), { tabWidth: 4, filepath: ".json" })
```

Comment:
Is `prettier` really needed here, if all it does is just suggest a bit of json to use?

Reply:
It's overkill for JSON but I used it because it's uncomplicated. It minimizes the diffs from the existing JSON that are suggested to the author.

Reply:
I'm not following this.  AFAICT, the above just formats the json value from scratch, so how is it minimizing diffs?
In addition to that, there's the `length` condition that makes it use a plain `stringify` for short objects, and prettier can't do *that* then it's even worse to use it for some cases and do its job manually in other cases.

Reply:
The format is closer to the files that are currently in the DT repo.

Reply:
I'll ask around whether people mind adding it...  But see also the comment below about using it only for some cases, which I still find disturbing, especially if prettier is included.

Reply:
I finally talked to @RyanCavanaugh about it, and he doesn't like adding prettier here too...

Reply:
Okay, thanks for asking around. I've removed it.